tests/DCPS/Thrasher/Publisher.cpp

```diff
@@ -64,22 +64,26 @@ Publisher::Publisher(const long domain_id, std::size_t samples_per_thread, bool
       throw std::runtime_error(" ERROR: create_datawriter failed!\n");
     }
     OpenDDS::DCPS::DataWriterImpl* wi = dynamic_cast<OpenDDS::DCPS::DataWriterImpl*>(dw_.in());
-    ACE_DEBUG((LM_INFO, (pfx_ + "  writer id: %C\n").c_str(), OpenDDS::DCPS::LogGuid(wi->get_repo_id()).c_str()));
+    if (wi) {
+      ACE_DEBUG((LM_INFO, (pfx_ + "  writer id: %C\n").c_str(), OpenDDS::DCPS::LogGuid(wi->get_repo_id()).c_str()));
+    } else {
+      throw std::runtime_error(" ERROR: dynamic_cast<OpenDDS::DCPS::DataWriterImpl*>(dw_.in()) failed!\n");
+    }
     writer_ = FooDataWriter::_narrow(dw_);
     if (!writer_) {
       throw std::runtime_error(" ERROR: FooDataWriter::_narrow failed!\n");
     }
     ACE_DEBUG((LM_INFO, (pfx_ + "->started\n").c_str()));
   } catch (const CORBA::Exception& e) {
-    e._tao_print_exception("caught in Publisher::svc()");
+    e._tao_print_exception(" ERROR: in Publisher::Publisher");
     cleanup();
     throw;
   } catch (const std::exception& e) {
     ACE_ERROR((LM_ERROR, (pfx_ + e.what()).c_str()));
```

Comment:
I'd make this one match the catch from Subscriber.cpp and have it add "ERROR" to the output here. Also, please add `#include <stdexcept>` to Thrasher.cpp Subscriber.cpp and Publisher.cpp